### PR TITLE
rename expression heads for type definitions to match keywords

### PR DESCRIPTION
**Pull request by JeffBezanson**

### CODE REVIEW

**Review by Sacha0 — COMMENTED** [review]
superficially lgtm! :)

**Comment by ararslan** [comment]
I don't suppose there's any way to deprecate the old expression heads? I know a couple packages are using `eval(Expr(:type, ...))` to define types in a (horrifically ugly) way that works with 0.5-0.7 without depwarns, e.g. JSON and I think Compat.

**Comment by JeffBezanson** [comment]
Yes, I can deprecate that particular use pattern.

**Comment by JeffBezanson** [comment]
Done. While I'm at it, I also changed the parsed representation to match the order that the subexpressions appear in the source.

File: NEWS.md [hunk 1]
```diff
@@ -35,6 +35,13 @@ Language changes
   * Nested `if` expressions that arise from the keyword `elseif` now use `elseif`
     as their expression head instead of `if` ([#21774]).
 
+  * Parsed and lowered forms of type definitions have been synchronized with their
+    new keywords ([#23157]). Expression heads are renamed as follows:
+    `type`           => `struct`
```
Comment (fredrikekre) [inline NEWS.md]: Perhaps add `*`s on this list with 4 space indent to render it as a bullet list? Current rendering looks a bit funny https://github.com/JuliaLang/julia/blob/acadf3acb6054adb657d6a9390616f2c1e792546/NEWS.md#language-changes

**Pull request merged**